### PR TITLE
segcache optimizations

### PR DESCRIPTION
**Pull request by brayniac**
Some optimizations for segcache performance. On a test system this
gives somewhere around 2% more performance. Benefit may vary
depending on config and workload.

- Reduce calls to Instant::recent() to reduce atomics in hot path.
- Remove trace logging in hot path.
- Change ordering of fields to optimize hot path.
- Use crossbeam_queue instead of crossbeam_channel to remove
  unnecessary channel level wake.

### CODE REVIEW

File: src/rust/storage/seg/src/hashtable/mod.rs [hunk 1]
```diff
@@ -557,7 +551,7 @@ impl HashTable {
 
                         if cas == get_cas(bucket.data[0]) {
                             // TODO(bmartin): what is expected on overflow of the cas bits?
-                            self.data[(hash & self.mask) as usize].data[0] += 1;
+                            self.data[(hash & self.mask) as usize].data[0] += 1 << 32;
```
Comment (thinkingfish) [inline src/rust/storage/seg/src/hashtable/mod.rs:554]: use `bucket_id as usize` and `CAS_BIT_SHIFT`?

File: src/rust/storage/seg/src/hashtable/mod.rs [hunk 2]
```diff
@@ -488,11 +484,11 @@ impl HashTable {
             insert_item_info = 0;
             self.data[bucket_id].data[N_BUCKET_SLOT - 1] = next_id as u64;
 
-            self.data[(hash & self.mask) as usize].data[0] += 0x0001_0000_0000_0000;
+            self.data[(hash & self.mask) as usize].data[0] += 0x0000_0000_0001_0000;
         }
 
         if insert_item_info == 0 {
-            self.data[(hash & self.mask) as usize].data[0] += 1;
+            self.data[(hash & self.mask) as usize].data[0] += 1 << 32;
```
Comment (thinkingfish) [inline src/rust/storage/seg/src/hashtable/mod.rs:491]: use `CAS_BIT_SHIFT`?

File: src/rust/storage/seg/src/hashtable/mod.rs (hunk 1 above)
Comment (brayniac) [reply src/rust/storage/seg/src/hashtable/mod.rs]: Good catch! Fixed

File: src/rust/storage/seg/src/hashtable/mod.rs (hunk 2 above)
Comment (brayniac) [reply src/rust/storage/seg/src/hashtable/mod.rs]: Fixed

**Review by thinkingfish — APPROVED** [review]

**Pull request merged**